### ISSUE TITLE
trouble adding OLED display

### ISSUE THREAD

**Issue by hjw** [Scott216/Weather_Station_Data#3]
Hi Scott,
I got your code up and running with an rfm69hcw breakout board and a redboard (UNO clone).. both from sparkfun. I commented out the ethernet and sensor related stuff and I can receive  data from my Davis station and print it out to the terminal console just fine. 
I'd like to add a small (micro_OLED breakout, also sparkfun) also on the SPI bus to the mix and have it display some of the weather data, but I've had no luck doing this and can't figure out why.
All of my connections seem correct and I can 1) upload code which talks to the rfm69 fine, 2) upload code which talks to the OLED fine and 3) talks to both (I think), but then it hangs / doesn't connect to the weather station. I am not changing any wiring in order to test these 3 scenarios, simply the uploaded code.
In scenario 3 the display inits and outputs a test image and then the terminal/console prints out the debug text:
Weather Station v0.51
Using regular Moteino
Reboots = 48
Free mem: 1023

And then nothing, no "No new data from ISS...." messages, no hopCount increase messages (which I put in to try to track where things are hanging), nothing. The radio normally takes around 2 minutes to connect with the weather station for the first time but if I init the OLED along with the radio it's sat around for 20 minutes without getting past the initial debug text. 
Any ideas? I'm using the default SS of 10 for the radio and pins 
#define OLED_PIN_RESET 9
#define OLED_PIN_DC  8
#define OLED_PIN_CS  5
for the OLED. 
Sincerely,
Hawley

**Comment by Scott216** [comment]
I'm guessing that both the radio and OLED are using the same SS/CS pin (D10).  If that's the case, it's probably the cause of your lockup.   To fix it you would have to modify either the RFM library or OLED library to use a different pin - this may not be that easy to do.  If your OLED breakout supports I2C, I'd try using that instead of SPI.

**Comment by hjw** [comment]
the radio is using ss/cs 10 and the OLED is using ss/cs 5.

Have you ever managed to add another component to the SPI bus along with the radio? All of the example of adding additional sensors etc add them to the I2C bus.

-Hawley

**Comment by Scott216** [comment]
Yes, I use an Ethernet board on SPI, but I had to get someone to modify the Ethernet library so it would use a different SS pin.  But my setup does lock up about once a week - not sure why.  I'm planning on using an Moteino Mega which has a lot more memory to see if that helps.  One problem I have now is Weather Underground changed something on their servers and I can't send them data anymore.  I haven't had time to look into that issue.
